### PR TITLE
More member exports from azure-devops

### PR DESCRIPTION
**Pull request by nmarinel**
## Hey, I just made a Pull Request!

My org is developing an internal plugin that leans on the `azure-devops` plugin, and it could use a few more exported members, in particular azureDevOpsApiRef and some react hooks. It looks like the `catalog-react` plugin and others make similar exports, which leads me to think this is safe and proper.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/azure-devops/src/index.ts [hunk 1]
```diff
@@ -25,6 +25,10 @@ export {
 } from './plugin';
 
 export { AzurePullRequestsIcon } from './components/AzurePullRequestsIcon';
```
Comment (awanlin) [inline plugins/azure-devops/src/index.ts:27]: Can we also export `AzurePipelinesIcon` while we are doing these exports for consistency sake?

**Review by awanlin — CHANGES_REQUESTED** [review]
As one of the owners/maintainers of this plugin I don't really have any issues with this as long as it's clear we may make breaking changes that we won't know how will impact you.

I would like to know why you would not want to contribute the changes that need these exports? Is this something that very specific to your org? I'm personally happy when I see people adding new features as it helps the rest of the community

**Comment by nmarinel** [comment]
@awanlin Absolutely! My org is working on improving our release management, which in this case means extending git tags with "target release" metadata stored in the backstage database. So it's convenient for us to use the azure-devops api to grab git tag info, and then we can manage the database dependency in our own plugin. Down the road though, we're hoping to leverage the Azure Devops API for release pipelines more which will likely lead to some enhancements appropriate for this plugin.

**Comment by awanlin** [comment]
@nmarinel, ok, cool, I was chatting with someone on Discord about the same thing a few weeks back.

As for the release pipelines can you maybe log an issue for that? We are starting to use the Multi-stage pipelines more as well as the Environments feature and I've been thinking of doing some work around that this fall. Having the issue would help line up our efforts!

**Review by awanlin — APPROVED** [review]

**Comment by nmarinel** [comment]
:D yep, that was me on Discord. I originally made internal copies of this plugin with the database dependency, but then once we realized what bad form that was and how easy it is to just make our own internal plugin for our unique business logic, we switched back to depending on this plugin properly. All part of the learning process :)

I'll log an issue like you said. Thanks for reviewing!

**Review by jhaals — COMMENTED** [review]
Hi!

Exports of the API and Client feels like totally valid use cases. However the other exports such as hooks and icons are tied to this plugin and having this part of the external api concerns me especially as they're very generically named making it hard to distinguish from other functionality in backstage. It's going to affect development in other parts of the project doing imports and getting suggestions for these azure hooks.
I suggest that we export the client and api and that you copy the other hooks you need to your internal package as the hooks seems to be fairly small.

**Comment by nmarinel** [comment]
@jhaals No problem, I can copy the icon and hooks where needed. I have pushed a new commit that removes the hooks and the newly exported icons

**Review by jhaals — APPROVED** [review]
Thanks for the adjustments @nmarinel, and @awanlin for the review! 👍 🏅

**Pull request merged**